components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridPolicyPersistenceManagerTest.java

```diff
@@ -0,0 +1,337 @@
+ */
+@WithCarbonHome
+@WithRegistry(injectToSingletons = {EntitlementServiceComponent.class})
+@WithRealmService(injectToSingletons = {EntitlementConfigHolder.class}, initUserStoreManager = true)
+@WithH2Database(jndiName = "jdbc/WSO2IdentityDB", files = {"dbscripts/h2.sql"})
+public class HybridPolicyPersistenceManagerTest extends PolicyPersistenceManagerTest {
+
+    private JDBCPolicyPersistenceManager jdbcPolicyPersistenceManager;
+    private RegistryPolicyPersistenceManager registryPolicyPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        setPrivateStaticField(PersistenceManagerFactory.class, "POLICY_STORAGE_TYPE", "hybrid");
+
+        Properties engineProperties = new Properties();
+        engineProperties.put(PDPConstants.MAX_NO_OF_POLICY_VERSIONS, "0");
+        EntitlementConfigHolder.getInstance().setEngineProperties(engineProperties);
+
+        Properties storeProps = new Properties();
+        policyPersistenceManager = new HybridPolicyPersistenceManager();
+        policyPersistenceManager.init(storeProps);
+        jdbcPolicyPersistenceManager = new JDBCPolicyPersistenceManager();
+        registryPolicyPersistenceManager = new RegistryPolicyPersistenceManager();
+        registryPolicyPersistenceManager.init(storeProps);
+    }
+
+    @Test(dataProvider = "policyStoreData", dependsOnMethods = {"testUpdatePDPPolicy"})
+    public void testGetPolicyIdentifiers(Object papPolicy1, Object pdpPolicy1, Object papPolicy2, Object pdpPolicy2,
+                                         Object papPolicy3, Object pdpPolicy3) throws Exception {
+
+        PolicyStoreDTO inputPDPPolicy1 = (PolicyStoreDTO) pdpPolicy1;
+        PolicyStoreDTO inputPDPPolicy2 = (PolicyStoreDTO) pdpPolicy2;
+        PolicyStoreDTO inputPDPPolicy3 = (PolicyStoreDTO) pdpPolicy3;
+
+        PolicyDTO inputPAPPolicy1 = (PolicyDTO) papPolicy1;
+        PolicyDTO inputPAPPolicy2 = (PolicyDTO) papPolicy2;
+        PolicyDTO inputPAPPolicy3 = (PolicyDTO) papPolicy3;
+
+        policyPersistenceManager.addOrUpdatePolicy(inputPAPPolicy1, true);
+        policyPersistenceManager.addOrUpdatePolicy(inputPAPPolicy2, true);
+        policyPersistenceManager.addOrUpdatePolicy(inputPAPPolicy3, true);
+
+        String[] policyIdentifiersBeforePublishing = ((HybridPolicyPersistenceManager) policyPersistenceManager)
+                .getPolicyIdentifiers();
+        assertEquals(policyIdentifiersBeforePublishing.length, 0);
+
+        policyPersistenceManager.addPolicy(inputPDPPolicy1);
+        policyPersistenceManager.addPolicy(inputPDPPolicy2);
+        policyPersistenceManager.addPolicy(inputPDPPolicy3);
+
+        String[] policyIdentifiersAfterPublishing = ((HybridPolicyPersistenceManager) policyPersistenceManager).
+                getPolicyIdentifiers();
+        assertEquals(policyIdentifiersAfterPublishing.length, 3);
+
+        policyPersistenceManager.deletePolicy(inputPDPPolicy1.getPolicyId());
```

Comment:
Yes. If the data was not cleared it will be affected to the remaining tests as well. So moved the data cleaning parts to a after method in the latest changes.